provisioning/service/samples/createdelete_sample.js

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var assert = require('chai').assert;
+
+var provisioningClient = require('azure-iot-provisioning-service-client');
+
+var serviceClient = provisioningClient.DeviceEnrollment.fromConnectionString(process.argv[2]);
+
+var enrollment1 = {
+  registrationId: 'first',
+  attestation: {
+    type: 'tpm',
+    tpm: {
+      endorsementKey: 'a',
+      storageRootKey: 'b'
+    }
+  }
+};
+
+var enrollment2 = {
+  registrationId: 'second',
+  attestation: {
+    type: 'tpm',
+    tpm: {
+      endorsementKey: 'a',
+      storageRootKey: 'b'
+    }
+  }
+};
+
+serviceClient.createOrUpdateIndividualEnrollment(enrollment1, function(err, firstEnrollmentResponse, httpResponse) {
+  if (!err) {
+    console.log("enrollment record returned for first: " + JSON.stringify(firstEnrollmentResponse, null, 2));
+    serviceClient.createOrUpdateIndividualEnrollment(enrollment2, function(err, secondEnrollmentResponse) {
+      if (!err) {
+        console.log("enrollment record returned for second: " + JSON.stringify(secondEnrollmentResponse, null, 2));
+        serviceClient.deleteIndividualEnrollment(firstEnrollmentResponse.registrationId, 'badeTag', function(err) {
+          assert.isNotNull(err, 'The first attempt to delete should have failed because of the bad etag');
+          console.log('First attempt to delete fails because a bad etag!  That is GOOD!!');
+          serviceClient.deleteIndividualEnrollment(firstEnrollmentResponse.registrationId, firstEnrollmentResponse.etag, function(err) {
+            if (!err) {
+              serviceClient.deleteIndividualEnrollment(secondEnrollmentResponse, function(err) {
+                if (err) {
+                  console.log('error deleting the second device: ' + JSON.stringify(err, null, 2));
+                }
+              });
+            } else {
+              console.log('error deleting the first device: ' + JSON.stringify(err, null, 2));
```

Comment:
>console [](start = 14, length = 7)

error handling almost always comes first - why change this?
also, to help readability it's better if shortest clauses come first. please reverse and use the usual if (err) else ...

Reply:
Done.

---
In reply to: [147065413](https://github.com/Azure/azure-iot-sdk-node/pull/121#discussion_r147065413) [](ancestors = 147065413)